src/components/timeline/TimelinePanel.svelte

```diff
@@ -14,15 +21,21 @@
   import TimelineViewControls from './TimelineViewControls.svelte';
 
   let timelineId: number = 0;
-  let isDeletingDirective: boolean = false;
 
   $: timeline = $view?.definition.plan.timelines.find(timeline => timeline.id === timelineId);
 
-  async function deleteActivityDirective() {
-    if (!isDeletingDirective) {
-      isDeletingDirective = true;
-      await effects.deleteActivityDirective($planId, $selectedActivity.id);
-      isDeletingDirective = false;
+  function deleteActivityDirective(event: CustomEvent<ActivityDirectiveId>) {
+    const { detail: activityDirectiveId } = event;
+    effects.deleteActivityDirective($planId, activityDirectiveId);
+  }
+
+  function onMouseDown(event: CustomEvent<MouseDown>) {
```

Comment:
Found an issue here where if you click on a directive and then click on a span the directive is deselected but the span is not selected. Happens when you go from a span to a directive as well.

Reply:
@duranb spent a ton of time on this today and made a fix. Thanks @duranb! Targeting a merge tomorrow.